pkg/services/baremetal/host/host.go

```diff
@@ -1053,6 +1057,10 @@ func (s *Service) actionProvisioning() actionResult {
 
 		isSSHTimeoutError, isSSHConnectionRefusedError, err := analyzeSSHOutputInstallImage(out, rescueSSHClient, portAfterInstallImage)
 		if err != nil {
+			if errors.Is(err, errUnexpectedHostName) {
```

Comment:
Sorry, this was not in the text of the PR.

Since I saw these places, I added a solution for "Detect if bm machine is owned by a different cluster".